### PR TITLE
Base: Add Keyboard Shortcuts & Tips and Tricks man pages + Keyboard symbols

### PR DESCRIPTION
**Pull request by cubiclove**
Add a list of system and general keyboard shortcuts as well as a list of tips and tricks to man7 (Miscellanea).
Add links to these in the Help-index to aid discoverability for new users.

Also add keyboard symbols to Katica Regular 12.
These symbols represent common icons found on a user's keyboard (PC and Mac). It will now be possible for these icons to be used in documentation (if desired) as Katica Regular 12 is used by LibMarkdown/the Help application.
Some bonus symbols have been added where they've been easy to make along the way.
I learnt about keyboard symbols [here](http://xahlee.info/comp/unicode_computing_symbols.html).

Glyph Demonstration Image

_It may look as if I've missed some symbols, such as some of the pointing triangles - that’s because there were already emojis in those codepoints_

https://www.unicode.org/charts/PDF/U1F100.pdf
🄰 - U+1F130 SQUARED LATIN CAPITAL LETTER A (Caps Lock - PC)

https://www.unicode.org/charts/PDF/U2190.pdf
← - U+2190 LEFTWARDS ARROW
↑ - U+2191 UPWARDS ARROW
→ - U+2192 RIGHTWARDS ARROW
↓ - U+2193 DOWNWARDS ARROW
↷ - U+21B7 CLOCKWISE TOP SEMICIRCLE ARROW (Redo)
↹ - U+21B9 LEFTWARDS ARROW TO BAR OVER RIGHTWARDS ARROW TO BAR (Tab)
⇞ - U+21DE UPWARDS ARROW WITH DOUBLE STROKE (Page Up - Mac)
⇟ - U+21DF DOWNWARDS ARROW WITH DOUBLE STROKE (Page Down - Mac)
⇤ - U+21E4 LEFTWARDS ARROW TO BAR
⇥ - U+21E5 RIGHTWARDS ARROW TO BAR (Tab)
⇦ - U+21E6 LEFTWARDS WHITE ARROW
⇧ - U+21E7 UPWARDS WHITE ARROW (Shift)
⇨ - U+21E8 RIGHTWARDS WHITE ARROW
⇩ - U+21E9 DOWNWARDS WHITE ARROW
⇪ - U+21EA UPWARDS WHITE ARROW FROM BAR (Caps Lock)
⇫ - U+21EB UPWARDS WHITE ARROW ON PEDESTAL
⇬ - U+21EC UPWARDS WHITE ARROW ON PEDESTAL WITH HORIZONTAL BAR (Caps Lock - French Canadian Layout Mac)
⇭ - U+21ED UPWARDS WHITE ARROW ON PEDESTAL WITH VERTICAL BAR (Number Lock - French Canadian Layout Mac)
⇮ - U+21EE UPWARDS WHITE DOUBLE ARROW (Option/Alt - French Canadian Layout Mac)
⇯ - U+21EF UPWARDS WHITE DOUBLE ARROW ON PEDESTAL
⇰ - U+21F0 RIGHTWARDS WHITE ARROW FROM WALL
⇱ - U+21F1 NORTH WEST ARROW TO CORNER
⇲ - U+21F2 SOUTH EAST ARROW TO CORNER
⇳ - U+21F3 UP DOWN WHITE ARROW

https://www.unicode.org/charts/PDF/U2200.pdf
⊞ - U+229E SQUARED PLUS (Used by [Wikipedia in place of a Windows key glyph](https://en.wikipedia.org/wiki/Windows_key))

https://www.unicode.org/charts/PDF/U2300.pdf
⌃ - U+2303 UP ARROWHEAD (Control - Mac)
⌘ - U+2318 PLACE OF INTEREST SIGN (Command - Mac)
⌤ - U+2324 UP ARROWHEAD BETWEEN TWO HORIZONTAL BARS (Numpad Enter - Mac)
⌥ - U+2325 OPTION KEY (Option/Alt - Mac)
⌦ - U+2326 ERASE TO THE RIGHT (Forward Delete - Mac)
⌧ - U+2327 X IN A RECTANGLE BOX (Clear - Mac)
⌫ - U+232B ERASE TO THE LEFT (Backspace - Mac)
⍰ - U+2370 APL FUNCTIONAL SYMBOL QUAD QUESTION (Help - Mac)
⎀ - U+2380 INSERTION SYMBOL
⎄ - U+2384 COMPOSITION SYMBOL (Compose Key - Sun Microsystems Keyboard)
⎆ - U+2386 ENTER SYMBOL (Enter - French Canadian Layout Mac)
⎈ - U+2388 HELM SYMBOL (Control - French Canadian Layout Mac)
⎋ - U+238B BROKEN CIRCLE WITH NORTHWEST ARROW (Escape - Mac)
⎌ - U+238C UNDO SYMBOL
⎗ - U+2397 PREVIOUS PAGE
⎘ - U+2398 NEXT PAGE
⎙ - U+2399 PRINT SCREEN SYMBOL (Print Screen - PC)
⎚ - U+239A CLEAR SCREEN SYMBOL
⏎ - U+23CE RETURN SYMBOL (Return/Enter)
⏻ - U+23FB POWER SYMBOL
⏼ - U+23FC POWER ON-OFF SYMBOL
⏽ - U+23FD POWER ON SYMBOL
⏾ - U+23FE POWER SLEEP SYMBOL

https://www.unicode.org/charts/PDF/U2460.pdf
ⓘ - U+24D8 CIRCLED LATIN SMALL LETTER I

https://www.unicode.org/charts/PDF/U25A0.pdf
▤ - U+25A4 SQUARE WITH HORIZONTAL FILL (Menu/App Key - PC)
▲ - U+25B2 BLACK UP-POINTING TRIANGLE
△ - U+25B3 WHITE UP-POINTING TRIANGLE
▷ - U+25B7 WHITE RIGHT-POINTING TRIANGLE
▼ - U+25BC BLACK DOWN-POINTING TRIANGLE
▽ - U+25BD WHITE DOWN-POINTING TRIANGLE
◁ - U+25C1 WHITE LEFT-POINTING TRIANGLE
◆ - U+25C6 BLACK DIAMOND (Meta Key - Sun Microsystems Type 6 Keyboard)
◇ - U+25C7 WHITE DIAMOND
◈ - U+25C8 WHITE DIAMOND CONTAINING BLACK SMALL DIAMOND

https://www.unicode.org/charts/PDF/U2700.pdf
✦ - U+2726 BLACK FOUR POINTED STAR (Hyper Key - Lisp Machine Keyboards)
✧ - U+2727 WHITE FOUR POINTED STAR (Hyper Key - Lisp Machine Keyboards)
✲ - U+2732 OPEN CENTRE ASTERISK (Control - PC)

https://www.unicode.org/charts/PDF/U27F0.pdf
⟳ - U+27F3 CLOCKWISE GAPPED CIRCLE ARROW (Reload - Chromebooks)
⟵ - U+27F5 LONG LEFTWARDS ARROW
⟶ - U+27F6 LONG RIGHTWARDS ARROW
⟷ - U+27F7 LONG LEFT RIGHT ARROW

https://www.unicode.org/charts/PDF/U2B00.pdf
⭘ - U+2B58 HEAVY CIRCLE (Power Off)

https://www.unicode.org/charts/PDF/U2900.pdf
⤒ - U+2912 UPWARDS ARROW TO BAR (Home - Mac)

### CODE REVIEW

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (konradekk) [inline Base/usr/share/man/man7/Tips-and-Tricks.md:57]: not sure if itʼs entirely correct, but iʼd suggest the following
```suggestion
* Make applications inspectable by including `MAKE_INSPECTABLE=1` in their environment. Inspectable processes can be examined with [Inspector](help://man/1/Applications/Inspector) via Remote Procedure Calls (RPCs). For example, to make the Eyes application inspectable, enter the following Shell command:
```
```sh
$ MAKE_INSPECTABLE=1 Eyes &
```

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (konradekk) [inline Base/usr/share/man/man7/Tips-and-Tricks.md:19]: i believe thatʼs not the context menu but the window menu (or at least should be—please double check that!).

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (konradekk) [inline Base/usr/share/man/man7/Tips-and-Tricks.md:10]: itʼd be good to use one of the “active” or “focus” words here for more clarity. 😉

```suggestion
* […], start typing to select the nearest match.
```

to be frank, “the nearest match” is not the best wording… any ideas to be just slightly more specific here? 🤔

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (konradekk) [inline Base/usr/share/man/man7/Tips-and-Tricks.md:14]: “compatible” is not the best here imho, it begs question for compatibility with what, and itʼs not really obvious that it is about the application itself; maybe this could be slightly more precise, e.g.  
```suggestion
* Many applications will open a file they can handle if you drag-and-drop it into their window.
```
i believe that the suggestion above is somewhat wrong and you could provide a better text for that hint. 🫣

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (konradekk) [inline Base/usr/share/man/man7/Tips-and-Tricks.md:31]: (“summon” is a word i really like but maybe this should be more of simple english—an “open” could be better here?)

Comment (konradekk) [inline Base/usr/share/man/man7/Tips-and-Tricks.md:31]: not really sure with that tip… it feels detached from the second one. i believe all tips should be standalone.

the first tip says in fact what the assistant is and how to open it with a keystroke…

hard to tell here, really!

i think that the second tip is a real tip; maybe you could rephrase the tips, so you could tell:
* how to open the assistant (dubious?)
* that you can open applications (more dubious?)
* that you can find files and open them in the default application
* that you can open urls in the default application (e.g. Browser)
* that can do simple computations in it (i believe itʼs possible)

files and urls could possibly be merged as those are virtually the same in some way…? 🤔

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (konradekk) [inline Base/usr/share/man/man7/Tips-and-Tricks.md:35]:
```suggestion
* Browser has built-in content filtering, which can be used for ad blocking. Update the filters in `~/.config/BrowserContentFiltering.txt`.
```
or something like that? my case is to point that it is the content filtering which is built in, not the ad blocking! 😉 (which might be a broader thing.)

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (konradekk) [inline Base/usr/share/man/man7/Tips-and-Tricks.md:38]: active voice?
```suggestion
* Create and edit custom keymaps with `$ KeyboardMapper`.
```

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (cubiclove) [reply Base/usr/share/man/man7/Tips-and-Tricks.md:31]: “Launch” or “Open” is probably the right word.
I liked “Summon” because it suggests Assistant is always available in the background ready to appear at a moments notice (though sounds a little occult).
I agree we should go with plain English rather than trying to be elegant.

**Comment by cubiclove** [comment]
@konradekk Thanks for your detailed suggestions. I will attend to them today.

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (konradekk) [reply Base/usr/share/man/man7/Tips-and-Tricks.md:31]: “bring up” is another thing that comes to my mind; but if we open dialog windows, then why not the assistant? “launch” is a synonym for “run” or “execute” for me—this would contradict your point about assistant being ready in the background (on the other hand, thatʼs a technicality which does not really matter for the end user 🤔).

(i believe that technically weʼre opening assistantʼs _window_. you could try to go around that if you want—pretty simple and very accurate).

**Comment by konradekk** [comment]
🎩

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (cubiclove) [reply Base/usr/share/man/man7/Tips-and-Tricks.md:31]: "Open" it is 🙂

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (cubiclove) [reply Base/usr/share/man/man7/Tips-and-Tricks.md:10]: I didn't want to use the world "search" because that has wider connotations and felt I'd found a good solution.
As you type only the icons/documents in your current focus (e.g. on the desktop or in the current folder in the File Manager) are being searched through.
I think your amendment makes it read much better though, so I will change it to that!

Comment (cubiclove) [reply Base/usr/share/man/man7/Tips-and-Tricks.md:10]: Or "When on the Desktop or in File Manager, start typing to select the nearest match shortcut or filename"

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (cubiclove) [reply Base/usr/share/man/man7/Tips-and-Tricks.md:14]: I've tried various alternatives ("will open a _relevant_ file?") but still think the original version works the best.
I think it's obvious you can drag a `.txt` file into Text Editor.

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (cubiclove) [reply Base/usr/share/man/man7/Tips-and-Tricks.md:31]: I'm trying to keep this short and sweet:

- The first tip quickly tells you there is a system search/launcher and how to access it.
- The second tips lists the only other functionality (that I know of).

I also thought you could do simple sums in Assistant, but have never managed it. Did we both dream about this functionality or was there a PR once upon a time (maybe it was never merged)?

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (cubiclove) [reply Base/usr/share/man/man7/Tips-and-Tricks.md:57]: This document is based on [tips.txt](https://github.com/SerenityOS/serenity/blob/f27f2e0d3b3fe08e68534676a58aab8e1eeea676/Base/home/anon/Documents/tips.txt) hence why this part is included even though I don't fully understand it.
I asked for help on #development and [Andrew Kaster taught me](https://discord.com/channels/830522505605283862/830739873119207426/1075928327711359036) how to actually use the command, so that's what I included. I tried your command and it didn't work for me - but I bow to you on this one, for I know not.

**Review by konradekk — APPROVED** [review]
time for another pair of eyes here! 🤗

**Comment by cubiclove** [comment]
@konradekk I think I found a solution to the trickiest wording:
"When on the Desktop or in File Manager, start typing and the nearest matching shortcut, file or folder will be selected."
I could shorten that to:
"When on the Desktop or in File Manager, start typing and the nearest matching _item_ will be selected" but that might not be clear enough (I'm starting to get a feel for this!)

**Comment by konradekk** [comment]
> @konradekk I think I found a solution to the trickiest wording: "When on the Desktop or in File Manager, start typing and the nearest matching shortcut, file or folder will be selected." I could shorten this to: "When on the Desktop or in File Manager, start typing and the nearest matching _item_ will be selected" but that might not be clear enough (I'm starting to get a feel for this!)

i think an “item” might be a thing there; yet thereʼs a problem with this “nearest matching”… i believe we should/could be more specific here. just checked the logic in `AbstractView.cpp` quickly: it is a case-insensitive prefix search. so maybe something like “[…], start typing the name of an item to select it”, or “[…], select an item by typing its name” (both assume you know that the thing you want to select exists), or “[…], select an item matching the text youʼve typed” (you want to perform some kind of a quick search/select).

the thing is that my all suggestions are more than clunky!

**Comment by cubiclove** [comment]
You're right. Our simple, final product:
"When on the Desktop or in File Manager, start typing the name of an item to select it."

Hopefully soon we are ready to merge
Freedom

**Comment by konradekk** [comment]
> Hopefully now we are ready to merge!
well, thatʼs not on me—let us wait for others! 😅 (esp. maintainers)

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (cubiclove) [reply Base/usr/share/man/man7/Tips-and-Tricks.md]: After testing multiple times, I've reverted to the original commands as they're the only ones that work:
```sh
$ export MAKE_INSPECTABLE=1
$ Eyes &
```

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (konradekk) [inline Base/usr/share/man/man7/Tips-and-Tricks.md:34]: i feel you can merge it with a previous (use full stop or semicolon to separate?).

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (konradekk) [inline Base/usr/share/man/man7/Tips-and-Tricks.md:33]: something along the lines `Evaluate simple mathematical expressions by prepending them with the equality sign`?

prefixes are strictly part of what your attaching them to; prepending separates this equality-sign computation marker from the expression logically—just as you put that in words. 🤔

need not to mimic that 100%, take it with a grain of salt (a bigger one) as i am not a native speaker! 🥸

File: Base/usr/share/man/man7/Tips-and-Tricks.md
Comment (konradekk) [inline Base/usr/share/man/man7/Tips-and-Tricks.md:47]: that would be a `file or folder path` to be precise. 😉 (“to copy it.”; not sure if mentioning the clipboard is necessary here)

**Comment by cubiclove** [comment]
The only other general keyboard shortcuts I've come across is `Ctrl+Backspace` for SIGQUIT, mentioned by Andreas [in this video](https://youtu.be/rRGu8DVrlHQ?t=1923).
I haven't added it to this list because I don't fully understand the functionality and haven't been able to get it to work. Is it meant to be like force quitting a process? Can you use it anywhere or only in the terminal?

**Comment by konradekk** [comment]
> The only other general keyboard shortcut I've come across is `Ctrl+Backspace` for `SIGQUIT`, mentioned by Andreas [in this video](https://youtu.be/rRGu8DVrlHQ?t=1923). I haven't added it to this documentation because I don't fully understand the functionality and haven't been able to get it to work. Is it meant to be like force quitting a process? Can you use it anywhere or only in the terminal?

`SIGQUIT` is one of the signals you can send to a process. i vaguely remember that sending through terminal is achieved with `Ctrl+\` but that needs a proper check. in general the signal is somewhat forceful way for user to stop the process; still not as forceful as other, maybe known to you, `SIGKILL` signal. you can definitely find some online resources on the exact details and can refer to the code base to understand the implementation details in serenity.

as you can see iʼve answered classic “it depends” to you 😅. going back where you started: no need to add things that you do not know/understand etc. — shall be good as is!

**Comment by cubiclove** [comment]
At first I thought "I've missed adding an essential keyboard shortcut (force quit an application/process)!" but now I see it's not exactly that. If this PR gets merged, someone who understands it could always add it in the right place down the line.

I'm starting to wonder if this PR isn't wanted (it would be nice to know upfront if that's the case), but it's more likely to be that PRs that drop off the first page mostly get forgotten about.
We can hope! I think it's in a really good initial state now.

**Comment by konradekk** [comment]
itʼs nice in my humble opinion — just try to advertise it a little (more?) on a proper discord channel? 🤔

**Review by ADKaster — APPROVED** [review]
This looks pretty cool, sorry no other maintainers have gotten to it yet! Like konrad mentioned, poking the #code-review channel is generally the best way but it seems like we're a bit low on maintainer energy this month (at least I am!). I would definitely suggest that if there's editorial nitpicks on a file in documentation to discuss them on discord or open a follow-up PR rather than leaving a ton of those types of comments on a review though. There's a balance.

**Comment by cubiclove** [comment]
Thanks Andrew!

You're right. In the future I will use Google Docs or an equivalent to get feedback on documentation and only submit the final draft.

**Pull request merged**
